src/events/events.ts

```diff
@@ -0,0 +1,391 @@
+    const logsID: string = server.logs;
+
+    // Skip if the server isn't subscribed to logs
+    if (!logsID) {
+        return;
+    }
+
+    const logs: TextChannel = <TextChannel> client.channels.cache.get(logsID);
+    const embed: EmbedBuilder = genMessageEditEmbed(before, after);
+
+    // If embed is null, the changes aren't significant to Sushi Bot
+    if (!embed) {
+        return;
+    }
+
+    try {
+        // Log to the logs channel
+        await logs.send({ embeds: [embed] });
+    } catch {
+        // Remove configurations if logging failed.
+        // Most likely causes are that the channel was deleted or that
+        // Sushi Bot had its permissions revoked
+        server.logs = null;
+        await server.save();
+    }
+};
+
+/**
+ * Logs message delete events to servers subscribed to logs.
+ * 
+ * @param client the Discord bot
+ * @param message the message that was deleted
+ */
+export const onMessageDelete = async (client: Client, message: Message): Promise<void> => {
+    // The server and logs channel ID of the server that had a message deleted
+    const server: Server = await Server.get(message.guild.id);
+    const logsID: string = server.logs;
+
+    // Skip if the server isn't subscribed to logs
+    if (!logsID) {
+        return;
+    }
+
+    const embed: EmbedBuilder = genMessageDeleteEmbed(message);
+
+    // If embed is null, the message wasn't significant to Sushi Bot
+    if (!embed) {
+        return;
+    }
+
+    const logs: TextChannel = <TextChannel> client.channels.cache.get(logsID);
+
+    try {
+        // Log to the logs channel
+        await logs.send({ embeds: [embed] });
+    } catch {
+        // Remove configurations if logging failed.
+        // Most likely causes are that the channel was deleted or that
+        // Sushi Bot had its permissions revoked
+        server.logs = null;
+        await server.save();
+    }
+};
+
+/**
+ * Triggered when the bot joins a Discord server.
+ * It leaves the server if the owner is ineligible to have the bot.
+ * 
+ * @param client the Discord bot
+ * @param server the Discord server joined
+ */
+export const onServerJoin = async (client: Client, server: Guild): Promise<void> => {
+    const eligible: boolean = await leaveIneligibleServer(client, server, getAdminLogsChannel());
+
+    if (eligible) {
+        const bl: Blacklist = await Blacklist.get();
+
+        for await (const [uid, reason] of bl.users) {
+            await server.bans.create(
+                uid,
+                { reason: reason }
+            );
+        }
+    }
+};
+
+/**
+ * Triggered when the bot leaves a Discord server,
+ * or when a Discord server is deleted.
+ * It deletes the server configurations from the database.
+ * 
+ * @param client the Discord bot
+ * @param server the Discord server left
+ */
+export const onServerLeave = async (client: Client, server: Guild): Promise<void> => {
+    await (await Server.get(server.id)).delete();
+};
```

Comment:
This gets triggered when Sushi Bot leaves a server
It deletes the server configurations for that server